Core/Inc/analyzer.h

```diff
@@ -45,25 +45,83 @@ extern const uint8_t POPULATED_THERM_LIST_H[NUM_THERMS_PER_CHIP];
 //#define MAX_SIZE_OF_HIST_QUEUE  300000U //bytes
 
 /**
- * @brief Pushes in a new data point if we have waited long enough
+ * @brief Calculate thermistor values and cell temps using thermistors.
+ * 
+ */
+void calc_cell_temps(acc_data_t *bmsdata);
+
+/**
+ * @brief Calculates pack temp, and avg, min, and max cell temperatures.
+ * 
+ */
+void calc_pack_temps(acc_data_t *bmsdata);
+
+/**
+ * @brief Calculate statistics about pack voltage, such as min and max cell volt, pack and avg voltage, pack and avg OCV, and deltas.
+ * 
+ */
+void calc_pack_voltage_stats(acc_data_t *bmsdata);
+
+/**
+ * @brief Calculate open cell voltages based on cell voltages and previous open cell voltages.
+ * 
+ */
+void calc_open_cell_voltage(acc_data_t *bmsdata);
+
+/**
+ * @brief Calculate cell resistances using cell temperatures and a cell temp to resistance lookup table.
+ * 
+ */
+void calc_cell_resistances(acc_data_t *bmsdata);
+
+/**
+ * @brief Calculate the discharge current limit based on OCV and cell resistance, and ensure that it is stable while driving.
+ * 
+ */
+void calc_dcl(acc_data_t *bmsdata);
+
+/**
+ * @brief Calculate the continuous discharge current limit based on cell temperatures and a cell temp to DCL lookup table.
+ * 
+ */
+void calc_cont_dcl(acc_data_t *bmsdata);
+
+/**
+ * @brief Calculate the continuous charging current limit based on cell temperatures and a cell temp to CCL lookup table.
+ * 
+ */
+void calc_cont_ccl(acc_data_t *bmsdata);
+
+/**
+ * @brief Disable unreliable thermistors.
+ * 
+ */
+void disable_therms(acc_data_t *bmsdata);
+
+/**
+ * @brief Calcuate state of charge by interpolation using an OCV to SOC lookup table.
+ * 
+ */
+void calc_state_of_charge(acc_data_t *bmsdata);
+
+/**
+ * @brief Calculate the percent of voltage readings that are noisy.
+ * 
+ */
+void calc_noise_volt_percent(acc_data_t *bmsdata);
+
```

Comment:
these are all internal functions; they should not be declared in the .h

Reply:
They're used in shep_tasks